src/Dhall.hs

```diff
@@ -434,20 +437,18 @@ Just 1
 maybe :: Type a -> Type (Maybe a)
 maybe (Type extractIn expectedIn) = Type extractOut expectedOut
   where
-    extractOut (OptionalLit _ es) = traverse extractIn es'
-      where
-        es' = if Data.Vector.null es then Nothing else Just (Data.Vector.head es)
-    extractOut _ = Nothing
+    extractOut (OptionalLit _ es) = traverse extractIn es
+    extractOut  _                 = Nothing
 
     expectedOut = App Optional expectedIn
 
-{-| Decode a `Vector`
-
->>> input (vector integer) "[1, 2, 3]"
+{-| Decode a `Seq`
+ -
+>>> input (sequence integer) "[1, 2, 3]"
 [1,2,3]
 -}
-vector :: Type a -> Type (Vector a)
-vector (Type extractIn expectedIn) = Type extractOut expectedOut
```

Comment:
You could still keep this `Type` and add `sequence`. Takes one breaking change out.

Reply:
It's still present (see export list), but I just relocated it below so that they were in logical order (i.e. `sequence` is the most direct one and then `list` is defined in terms of `sequence` and `vector` is defined in terms of `list`)

Reply:
Ah cool, thanks!